kernel/aster-nix/src/syscall/stat.rs

```diff
@@ -75,38 +75,6 @@ pub fn sys_fstatat(
     Ok(SyscallReturn::Return(0))
 }
 
-/// File type mask.
-const S_IFMT: u16 = 0o170000;
-
-/// Enum representing different file types.
-#[derive(Debug, PartialEq)]
-pub enum FileType {
-    Socket,
```

Comment:
I'm basically fine with removing redundant `FileType` or `InodeType` since they look exactly the same. But two remaining concerns:
1. In Linux, are there any differences between `struct stat. umode_t mode` in `include/linux/stat.h` and `struct inode. umode_t i_mode` in `include/linux/fs.h`? I believe we will have more confidence in this removal after we clear that out.
2. If we decide to stick to `InodeType`, all alonged variable and function names should be consistent, `let file_type` or `file_type()` should not appeared.

Reply:
> 1. In Linux, are there any differences between `struct stat. umode_t mode` in `include/linux/stat.h` and `struct inode. umode_t i_mode` in `include/linux/fs.h`? I believe we will have more confidence in this removal after we clear that out.

I think there are no differences between `mode` and `i_node`. I have checked the implementation of ramfs and see the [following code](https://github.com/torvalds/linux/blob/master/fs/inode.c#L2421). It shows that Linux sets the `i_mode` to the `mode` from `mknod` directly. The call chain:

```txt
1. fs/namei.c: do_mknodat
2. fs/namei.c: vfs_mknod
3. fs/ramfs/inode.c: ramfs_mknod
4. fs/ramfs/inode.c: ramfs_get_inode
5. fs/inode.c: init_special_inode
```